data/src/test/java/org/apache/iceberg/data/GenericAppenderHelper.java

```diff
@@ -73,13 +74,12 @@ public DataFile writeFile(StructLike partition, List<Record> records) throws IOE
     Preconditions.checkNotNull(table, "table not set");
     File file = tmp.newFile();
     Assert.assertTrue(file.delete());
-    return appendToLocalFile(table, file, fileFormat, partition, records);
+    return appendToLocalFile(table, file, fileFormat, partition, records, table.spec());
   }
 
-  private static DataFile appendToLocalFile(
-      Table table, File file, FileFormat format, StructLike partition, List<Record> records)
-      throws IOException {
-    FileAppender<Record> appender = new GenericAppenderFactory(table.schema()).newAppender(
+  private static DataFile appendToLocalFile(Table table, File file, FileFormat format, StructLike partition,
+                                            List<Record> records, PartitionSpec spec) throws IOException {
+    FileAppender<Record> appender = new GenericAppenderFactory(table.schema(), spec).newAppender(
```

Comment:
Could we keep the constructor this was using before so we don't need to change any tests that only use `newAppender`?

Reply:
I think I've counted at least 10 files that would not need to change if we kept the original constructor.